src/Umbraco.Web.UI.Client/src/common/services/tour.service.js

```diff
@@ -1,455 +1,200 @@
-                        title: "Properties",
-                        content: "Like the content section you can also find default properties about the media item. You will find these under the properties tab.",
-                        event: "click"
-                    },
-                    {
-                        element: "[data-element='editor-media'] [data-element='property-_umb_urls']",
-                        title: "Link to media",
-                        content: "The path to the media item..."
-                    },
-                    {
-                        element: "[data-element='editor-media'] [data-element='property-_umb_updatedate']",
-                        title: "Last edited",
-                        content: "...and information about when the media item has been created and edited."
-                    }
-                ]
-            }
-        ];
+        /**
+         * @ngdoc method
+         * @name umbraco.services.tourService#registerTour
+         * @methodOf umbraco.services.tourService
+         *
+         * @description
+         * Registers a tour in the service
+		 * @param {Object} tour The tour you want to register in the service
+         * @param {String} tour.name The tour name
+         * @param {String} tour.alias The tour alias
+         * @param {Array} tour.steps Array of tour steps
+         * @param {String} tour.step.title Step title
+         * @param {DomElement} tour.step.content Step content (pass in any HTML markup)
+         * @param {DomElement} tour.step.element Highlight a DOM-element
+         * @param {Boolean} tour.step.elementPreventClick Adds invisible layer on top of highligted element to prevent all clicks and interaction with it
+         * @param {Number} tour.step.backdropOpacity Sets the backdrop opacity (default 0.4)
+		 */
+        function registerTour(newTour) {
+            tours.push(newTour);
+            eventsService.emit("appState.tour.updatedTours", tours);
+        }
+
+        /**
+         * @ngdoc method
+         * @name umbraco.services.tourService#registerTours
+         * @methodOf umbraco.services.tourService
+         *
+         * @description
+         * Registers an array of tours in the service
+		 * @param {Array} tours The tours to register in the service
+		 */
+        function registerTours(newTours) {
+            angular.forEach(newTours, function(newTour){
+                tours.push(newTour);
+            });
+            eventsService.emit("appState.tour.updatedTours", tours);
+        }
+
+        /**
+         * @ngdoc method
+         * @name umbraco.services.tourService#unregisterTour
+         * @methodOf umbraco.services.tourService
+         *
+         * @description
+         * Unregisters a tour in the service
+		 * @param {String} tourAlias The tour alias of the tour you want to unregister
+		 */
+        function unregisterTour(tourAlias) {
+            tours = tours.filter(function( obj ) {
```

Comment:
@madsrasmussen the small problem with this is that we don't seem to have any validation or checks when registering a tour, by pushing the object into the array, to verify that we don't already have a tour registered with that alias.

As this will could remove more than one tour if there are multiple in the array with the same alias